modules/sdk-coin-avaxp/src/lib/transactionBuilderFactory.ts

```diff
@@ -17,19 +19,46 @@ export class TransactionBuilderFactory extends BaseTransactionBuilderFactory {
   /** @inheritdoc */
   from(raw: string): TransactionBuilder {
     utils.validateRawTransaction(raw);
-    const tx = new Tx();
-    tx.fromBuffer(BufferAvax.from(raw, 'hex'));
+    let tx: Tx | EVMTx;
     let transactionBuilder: TransactionBuilder;
-    if (ValidatorTxBuilder.verifyTxType(tx.getUnsignedTx().getTransaction())) {
-      transactionBuilder = this.getValidatorBuilder();
-    } else if (ExportTxBuilder.verifyTxType(tx.getUnsignedTx().getTransaction())) {
-      transactionBuilder = this.getExportBuilder();
-    } else if (ImportTxBuilder.verifyTxType(tx.getUnsignedTx().getTransaction())) {
-      transactionBuilder = this.getImportBuilder();
-    } else {
-      throw new NotSupported('Transaction cannot be parsed or has an unsupported transaction type');
-    }
+    try {
+      tx = new Tx();
+      // could throw an error if a txType doesn't match.
+      tx.fromBuffer(BufferAvax.from(raw, 'hex'));
+
+      const sameBlockchainId =
+        utils.cb58Encode(tx.getUnsignedTx().getTransaction().getBlockchainID()) ===
+        (this._coinConfig.network as AvalancheNetwork).blockchainID;
+      if (!sameBlockchainId) {
+        throw new Error('It is not a transaction of this network');
+      }
```

Comment:
Duplicated code down in ln:49. This validation could be done inside a different try/catch block do avoid duplication.

Reply:
Please re-write this to remove code duplication. You can split it into 2 try/catch blocks.